Userland/Libraries/LibGfx/PNGWriter.cpp

```diff
@@ -155,13 +166,13 @@ union [[gnu::packed]] Pixel {
 };
 static_assert(AssertSize<Pixel, 4>());
 
-void PNGWriter::add_IDAT_chunk(Gfx::Bitmap const& bitmap)
+ErrorOr<void> PNGWriter::add_IDAT_chunk(Gfx::Bitmap const& bitmap)
 {
     PNGChunk png_chunk { "IDAT" };
-    png_chunk.reserve(bitmap.size_in_bytes());
+    TRY(png_chunk.reserve(bitmap.size_in_bytes()));
 
     ByteBuffer uncompressed_block_data;
-    uncompressed_block_data.ensure_capacity(bitmap.size_in_bytes() + bitmap.height());
+    TRY(uncompressed_block_data.try_ensure_capacity(bitmap.size_in_bytes() + bitmap.height()));
 
     Pixel dummy_scanline[bitmap.width()];
```

Comment:
How is dummy_scanline going to be allocated? Can it oom?

Reply:
That's a VLA, which cannot signal OOM. We could use a heap-allocated buffer instead, which might be a good idea since the array size is caller-controlled.